pwsh/Test-GitShaFiles.ps1

```diff
@@ -0,0 +1,38 @@
+<#
+.SYNOPSIS
+  Find files matching the given filter and compare their contents with the
+  given SHA. Throw an error if the SHA differs from any of the found files
+  contents.
+.DESCRIPTION
+  Use "Get-Help Test-GitShaFiles.ps1 -Detailed" for parameter descriptions.
+.LINK
+  https://github.com/pace-neutrons/ci-release-deploy
+#>
+param (
+  # The required Git SHA to match to the discovered files
+  [string]$RequiredSHA=$(throw "Mandatory argument 'RequiredSHA' not specified."),
+  # The filter to use to discover SHA files (this is passed to Get-ChildItem)
```

Comment:
This comment isn't complete; "the filter": what sort of filter? applied to what?
it's a filename filter for the file(s) containing a GitSHA, e.g. *git-revision*